manim/mobject/numbers.py

```diff
@@ -143,3 +145,76 @@ class Integer(DecimalNumber):
 
     def get_value(self):
         return int(np.round(super().get_value()))
+
+
+class Variable(VMobject):
+    """A class for displaying text that continuously updates to reflect the value of a python variable.
+
+    Automatically adds the text for the label and the value when instantiated and added to the screen.
+
+    Parameters
+    ----------
+    var : Union[:class:`int`, :class:`float`]
+        The python variable you need to keep track of and display.
+    label : Union[:class:`str`, :class:`~.Tex`, :class:`~.MathTex`, :class:`Text`]
+        The label for your variable, for example `x = ...`. To use math mode, for e.g.
+        subscripts, superscripts, etc. simply pass in a raw string.
+    var_type : Union[:class:`DecimalNumber`, :class:`Integer`], optional
+        The class used for displaying the number. Defaults to :class:`DecimalNumber`.
+    num_decimal_places : :class:`int`, optional
+        The number of decimal places to display in your variable. Defaults to 2.
+        If `var_type` is an :class:`Integer`, this parameter is ignored.
+    kwargs : Any
+            Other arguments to be passed to `~.Mobject`.
+
+    Attributes
+    ----------
+    label : Union[:class:`str`, :class:`~.Tex`, :class:`~.MathTex`, :class:`Text`]
+        The label for your variable, for example `x = ...`.
+    tracker : :class:`~.ValueTracker`
+        Useful in updating the value of your variable on-screen.
+    value : Union[:class:`DecimalNumber`, :class:`Integer`]
+        The tex for the value of your variable.
+
+    Examples
+    --------
+    Normal usage::
+        # DecimalNumber type
+        var = 0.5
+        on_screen_var = Variable(var, "var", num_decimal_places=3)
+        # Integer type
+        int_var = 0
+        on_screen_int_var = Variable(int_var, "int_var", var_type=Integer)
+        # Using math mode for the label
+        on_screen_int_var = Variable(int_var, "${a}_{i}$", var_type=Integer)
+
+    """
+
+    def __init__(
+        self, var, label, var_type=DecimalNumber, num_decimal_places=2, **kwargs
+    ):
+
+        the_class = type(label)
+        if the_class == str:
+            self.label = MathTex(label)
+        else:
+            self.label = label
+
```

Comment:
```suggestion
        self.label = MathTex(label) if isinstance(label,str) else label
```
Suggestion: I think this is a tad easier to read, and it doesn't make use of one-time variables like `the_class`.

Reply:
Agreed